backend/src/apiserver/resource/resource_manager_util.go

```diff
@@ -228,3 +229,28 @@ func OverrideParameterWithSystemDefault(workflow util.Workflow, apiRun *api.Run)
 	}
 	return nil
 }
+
+// Convert PipelineId in PipelineSpec to the pipeline's default pipeline version.
+// This is for legacy usage of pipeline id to create run. The standard way to
+// create run is by specifying the pipeline version.
```

Comment:
you mean to Latest Pipeline Version, right?

Reply:
It's actually default version. Usually the default is the latest. But we reserve the possibility to add feature such as allowing users to set a default version.